### PR TITLE
Update opentelemetry-concepts.mdx

### PR DESCRIPTION
**Pull request by jonahaapala**
Please follow [conventional commit standards](https://www.conventionalcommits.org/en/v1.0.0/)
in your commit messages and pull request title.

## Give us some context

* What problems does this PR solve?
Update OTEL docs to indicate support for non-nested homogenous array attributes. This is currently supported by the OTLP Metrics Consumer and should soon be true for Logs and Traces. This should go out with any updates to documentation for array support in NRDB which I believe is coming soon.

* Add any context that will help us review your changes such as testing notes,
  links to related docs, screenshots, etc.

🎉 
image

Like I said, we are still waiting on the full NRDB support but that should be landing soon.

### CODE REVIEW

**Comment by bradleycamacho** [comment]
Hey there @jonahaapala , thank for reporitng this! I have a doc jira issue (DOC-7796) to get the updates for the arrays feature staged for release. I'll be sure to publish this along side that.

File: src/content/docs/more-integrations/open-source-telemetry-integrations/opentelemetry/introduction-opentelemetry-new-relic.mdx [hunk 1]
```diff
@@ -265,8 +265,8 @@ Here are the OpenTelemetry data types we support and their associated mappings.
       <td>❌</td>
     </tr>
     <tr>
-      <td>[Array of primitives (homogeneous)](https://github.com/open-telemetry/opentelemetry-specification/blob/e13990125809a4ecc482553c32d05e38bb414acb/specification/common/common.md#attributes)</td>
-      <td>❌</td>
+      <td>[Array of primitives (homogeneous)](https://github.com/open-telemetry/opentelemetry-specification/blob/e13990125809a4ecc482553c32d05e38bb414acb/specification/common/common.md#attributes). Must be non-nested and 64 or less elements.</td>
+      <td>✅</td>
```
Comment (jonahaapala) [inline src/content/docs/more-integrations/open-source-telemetry-integrations/opentelemetry/introduction-opentelemetry-new-relic.mdx:269]: I suspect that we will also want to update the Logs and Traces sections, but other teams are working on that so I'll let them add whatever details they need.

Comment (bmcfeely) [reply src/content/docs/more-integrations/open-source-telemetry-integrations/opentelemetry/introduction-opentelemetry-new-relic.mdx:269]: Logging will have one order of the same docs, please!

File: src/content/docs/more-integrations/open-source-telemetry-integrations/opentelemetry/introduction-opentelemetry-new-relic.mdx [hunk 2]
```diff
@@ -305,12 +305,12 @@ For more details, see the Logs section of our [best practices](/docs/integration
     </tr>
     <tr>
       <td>Array messages</td>
-      <td>❌</td>
+      <td>✅</td>
```
Comment (bmcfeely) [inline src/content/docs/more-integrations/open-source-telemetry-integrations/opentelemetry/introduction-opentelemetry-new-relic.mdx:308]: I think this should have the same non-nested / homogeneous warning as attributes 🤔

Comment (bmcfeely) [reply src/content/docs/more-integrations/open-source-telemetry-integrations/opentelemetry/introduction-opentelemetry-new-relic.mdx:308]: Gah, where are my manners? Thank you! 😄

**Pull request merged**